Can't open any transaccion (crash)

**Describe the bug**
On a existing DB I use daily, opening any transaction crashes the App.

**To Reproduce**
Steps to reproduce the behavior:
1. Go to 'any account'
2. Click on 'any existing transaction'
3. Application crashes

**Expected behavior**
Just open the transactions, edit, create new ones, etc. App become unusable.

**Screenshots**

**Device Information:**
 - Device: Samsung A53
 - OS: Android 14
 - Storage: Google Drive
 - Synchronization: Sync on Start
 - Encrypted: No
 - App Version: Affected versions: 2024.04.15.1027 & 2024.04.16.1028. Rollback to 2024.04.01.1026 solved the issue.

**Additional context**
Full App reinstall + data delete and cache clear do not solved the issue.

I believe I have the same problem. Here is the DEBUG logcat of the session. 
I clicked on a transaction at around 10:40:15.

This logcat exerpt is obtained by clicking on the "share" button in the logcat window. The big LOGCAT button on the info page triggers a "Error: Failure from system" message.

I'll see during the day if I can make a  tiny dataset that triggers the crash.

E/ 04-21 10:40:09.649 oney.manager.ex(16986): Not starting debugger since process cannot load the jdwp agent.
E/ 04-21 10:40:09.818 QT      (16986): [QT]file does not exist
E/ 04-21 10:40:09.825 OpenGLRenderer(16986): Device claims wide gamut support, cannot find matching config, error = EGL_SUCCESS
W/ 04-21 10:40:09.825 OpenGLRenderer(16986): Failed to initialize 101010-2 format, error = EGL_SUCCESS
W/ 04-21 10:40:10.035 OpenGLRenderer(16986): Unknown dataspace 0
E/ 04-21 10:40:10.078 FBI     (16986): Can't load library: dlopen failed: library "libmagtsync.so" not found
W/ 04-21 10:40:13.774 WindowOnBackDispatcher(16986): OnBackInvokedCallback is not enabled for the application.
W/ 04-21 10:40:13.774 WindowOnBackDispatcher(16986): Set 'android:enableOnBackInvokedCallback="true"' in the application manifest.
E/ 04-21 10:40:16.279 Android-Iconics(16986): Can't init: com.mikepenz.mmex_icon_font_typeface_library.MMEXIconFont
E/ 04-21 10:40:16.438 AndroidRuntime(16986): FATAL EXCEPTION: main
E/ 04-21 10:40:16.438 AndroidRuntime(16986): Process: com.money.manager.ex, PID: 16986
E/ 04-21 10:40:16.438 AndroidRuntime(16986): java.lang.RuntimeException: Unable to start activity ComponentInfo{com.money.manager.ex/com.money.manager.ex.transactions.CheckingTransactionEditActivity}: java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 81
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4164)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:4322)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2685)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.os.Handler.dispatchMessage(Handler.java:106)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.os.Looper.loopOnce(Looper.java:230)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.os.Looper.loop(Looper.java:319)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.app.ActivityThread.main(ActivityThread.java:8893)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at java.lang.reflect.Method.invoke(Native Method)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:608)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1103)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): Caused by: java.lang.StringIndexOutOfBoundsException: begin 0, end -1, length 81
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at java.lang.String.checkBoundsBeginEnd(String.java:4466)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at java.lang.String.substring(String.java:2517)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at com.money.manager.ex.transactions.EditTransactionCommonFunctions.initAttachmentControls(EditTransactionCommonFunctions.java:736)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at com.money.manager.ex.transactions.CheckingTransactionEditActivity.initializeInputControls(CheckingTransactionEditActivity.java:383)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at com.money.manager.ex.transactions.CheckingTransactionEditActivity.onCreate(CheckingTransactionEditActivity.java:122)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.app.Activity.performCreate(Activity.java:8944)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.app.Activity.performCreate(Activity.java:8913)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1456)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:4146)
E/ 04-21 10:40:16.438 AndroidRuntime(16986): 	... 12 more
E/ 04-21 10:40:26.684 oney.manager.ex(17305): Not starting debugger since process cannot load the jdwp agent.

--- a/app/build.gradle
+++ b/app/build.gradle
@@ -27,9 +27,9 @@ android {
 
     defaultConfig {
         applicationId "com.money.manager.ex"
-        versionCode 1028
+        versionCode 1029
         //versionName getVersionAsDate() + "." + versionCode
-        versionName "2024.04.16"
+        versionName "2024.04.21"
         // $applicationId
         setProperty("archivesBaseName", "ammx-$versionName")
 

--- a/app/src/main/java/com/money/manager/ex/transactions/EditTransactionCommonFunctions.java
+++ b/app/src/main/java/com/money/manager/ex/transactions/EditTransactionCommonFunctions.java
@@ -733,7 +733,7 @@ public class EditTransactionCommonFunctions {
         String attachmentsFolder = String.format("MMEX_%s_Attachments", file.getName().substring(0, file.getName().lastIndexOf('.')));
 
         String remotePath = mDatabases.get().getCurrent().remotePath;
-        String baseUri = remotePath.substring(0, remotePath.lastIndexOf(file.getName()));
+        String baseUri = remotePath.substring(0, remotePath.lastIndexOf("%2F") + 3);
         for (Attachment att : getAttachments()) {
             String uri = baseUri +  attachmentsFolder + "%2F" + att.getRefType() + "%2F" + att.getFilename();
 

--- a/fastlane/metadata/android/en-US/changelogs/1029.txt
+++ b/fastlane/metadata/android/en-US/changelogs/1029.txt
@@ -1,0 +1,1 @@
+- bugfix: crash for database file folder retrieval

